File: HelpForTeacher (Java)/app/src/main/java/com/teacherhelper/TeacherApp.java
package com.teacherhelper;

import android.app.Application;
import android.support.multidex.MultiDexApplication;

import com.kinvey.android.Client;

/**
 * Created by Prots on 3/15/16.
 */
public class TeacherApp extends Application {

    Client sharedClient;

    @Override
    public void onCreate() {
        super.onCreate();
        sharedClient = new Client.Builder(this).build();
    }

    public Client getSharedClient(){
        return sharedClient;
    }
}

File: HelpForTeacher (Kotlin)/app/src/main/java/com/teacherhelper/TeacherApp.kt
package com.teacherhelper

import android.app.Application
import android.support.multidex.MultiDexApplication

import com.kinvey.android.Client

class TeacherApp : Application() {

    var sharedClient: KniveyClient?  = null
        internal set

    override fun onCreate() {
        super.onCreate()
        sharedClient = KniveyClient(Client.Builder(this).build())
    }
}
